projects/igniteui-angular/src/lib/calendar/calendar.component.ts

```diff
@@ -296,6 +314,59 @@ export class IgxCalendarComponent implements OnInit, ControlValueAccessor {
      */
     @Input()
     public vertical = false;
+
+    /**
+     * Gets the disabled dates descriptors.
+     * ```typescript
+     * let disabledDates = this.calendar.disabledDates;
+     * ```
+     */
+    public get disabledDates(): DateRangeDescriptor[] {
+        return this._disabledDates;
+    }
+
+    /**
+     * Sets the disabled dates' descriptors.
+     * ```typescript
+     *@ViewChild("MyCalendar")
+     *public calendar: IgCalendarComponent;
+     *ngOnInit(){
+     *    this.calendar.disabledDates = [
+     *      new DateRangeDescriptor(DateRangeType.Between, [new Date("2020-1-1"), new Date("2020-1-15")]),
+     *      new DateRangeDescriptor(DateRangeType.Weekends)];
+     *}
+     *```
+     */
+    public set disabledDates(value: DateRangeDescriptor[]) {
+        this._disabledDates = value;
+    }
+
+    /**
+     * Gets the special dates descriptors.
+     * ```typescript
+     * let specialDates = this.calendar.specialDates;
+     * ```
+     */
+    public get specialDates(): DateRangeDescriptor[] {
```

Comment:
Let's think about the name of this `@Input`. Off the top of my head: 'holidays', 'highlightDates`, `significantDates`... Throw in your suggestions as well!

Reply:
Occasions, occasionDates, otherOccasions, otherDates, observanceDates

Reply:
IMHO `specialDates` is an excellent name.

Reply:
@tiliev I'm actually also starting to think that the name is good as is.

Reply:
@kdinev, @tiliev, lets keep that name as it is. After making a little research it seems like a common usage for - Holidays, PTO days, National holidays, teambuildings, key days and any other Day of the year with special meaning.